onnxruntime/test/mlas/unittest/test_qgemm.cpp

```diff
@@ -66,9 +85,18 @@ static size_t QGemmRegistShortExecute() {
       count += QgemmShortExecuteTest<uint8_t, int8_t, float, true, true>::RegisterShortExecuteTests();
       count += QgemmShortExecuteTest<uint8_t, int8_t, int32_t, true, true>::RegisterShortExecuteTests();
     }
-    if (MlasGemmPackBSize(128, 128, true /*AIsSigned*/, true /*BIsSigned*/) > 0) {
-      count += QgemmShortExecuteTest<int8_t, int8_t, float, true, true>::RegisterShortExecuteTests();
-      count += QgemmShortExecuteTest<int8_t, int8_t, int32_t, true, true>::RegisterShortExecuteTests();
+    try {
+      if (MlasGemmPackBSize(128, 128, true /*AIsSigned*/, true /*BIsSigned*/) > 0) {
+        count += QgemmShortExecuteTest<int8_t, int8_t, float, true, true>::RegisterShortExecuteTests();
+        count += QgemmShortExecuteTest<int8_t, int8_t, int32_t, true, true>::RegisterShortExecuteTests();
+      }
+      if (MlasGemmPackBSize(128, 128, true /*AIsSigned*/, false /*BIsSigned*/) > 0) {
+        count += QgemmShortExecuteTest<int8_t, uint8_t, float, true, false>::RegisterShortExecuteTests();
+        count += QgemmShortExecuteTest<int8_t, uint8_t, int32_t, true, false>::RegisterShortExecuteTests();
+      }
+    } catch (const std::invalid_argument& e) {
+      (void)e;
```

Comment:
Is the exception introduced by your PR?

Reply:
Not directly. 
The function MlasGemmQuantGetDispatch throws the error if the platform does not support the target signed/unsigned combination. Original code seems to fallback to C++ implementation by default for U8X8 or X8S8. When I added the new dispatchers I had originally set them to nullptr, then had to add these try/catch blocks when testing different platforms that couldn't support S8S8/S8U8.
I was unsure whether Microsoft wants all implementation routes to fallback to default C++ implementation if no other option is available. At some point I changed my dispatchers to be initialized to the default implementation so there would always be something functional. I believe with this initialization, it is unnecessary to have the try/catch blocks as it will never be null, but I wanted to get the intention clarified.

After reviewing some of the automated build results, I see that exceptions are not desired/enabled for certain platforms, so I will leave the dispatchers initialized to the default C++ method and remove the exception handling.